### ISSUE TITLE
media-type-examples-override won't work with json containing 'operationId'

### ISSUE THREAD

**Issue by arnaudroubinet** [Redocly/redocly-cli#1817]
**Describe the bug**

Using media-type-examples-override, having in your example a field call operationId break the example replacement.

I'm sorry, I don't know how to add correct yaml into github issue.

**Expected behavior**
Example with operationId variable is managed

**Redocly Version(s)**
Docker image [1.25.15](https://hub.docker.com/layers/redocly/cli/1.25.15/images/sha256-a9dc885914b2896c3637ea54425b496d8dd61869407f8578c77771476e372ba0?context=explore)

**To Reproduce**

media-type-examples-override:
-> operationIds:
-> -> myEndpoint:
-> -> -> request:
-> -> -> -> application/mymediatype-1+json: /spec/examples/mymediatype/mymediatype-request.yaml

---
#Acceptable mymediatype-request.yaml content
WorkingExample:
-> summary: MyExampleSummary
-> value: {"operationzId": "myoperation"}

---
#Refused mymediatype-request.yaml content
BreakingExample:
-> summary: MyExampleSummary
-> value: {"operationId": "myoperation"}

---
#Bundle configuration content
[example.txt](https://github.com/user-attachments/files/18041986/example.txt)

**Comment by arnaudroubinet** [comment]
Moral: never rush writing an anomaly...
The problem is with the filter-in decorator.

[openapi - bug.zip](https://github.com/user-attachments/files/18362387/openapi.-.bug.zip)

You can see in openapi-updated that the schematic is truncated for myOperation.

**Comment by tatomyr** [comment]
Well, that's an interesting coincidence. The way `filter-in` works, it looks through all the document nodes and removes properties on ALL levels where it finds the target property (in your case it happens to be `operationId` which is also present in one of the examples) that aren't equal to the specified values. This is the intended behaviour, so I'm closing the issue.

What you can do to achieve your goal: 
- use [filter-out](https://redocly.com/docs/cli/decorators/filter-out) decorator instead (it will work much more predictable as it removes only specified cases, and there are little chances you run into a similar collision)
- write a [custom decorator using plugins](https://redocly.com/docs/cli/custom-plugins/custom-decorators)
- use a different target property

I hope that helps.

**Comment by arnaudroubinet** [comment]
My use case is to extract specific operations for the documentation of my integration team.

Using filter-out will force me to initialize and maintain the filter for hundreds of operations instead of specifying the few what I want to see.

I think a filter functionality with a match instead of a property (so as not to break retro compatibility) could be an interesting improvement.

For example : 

decorators:
  filter-in:
    matchProperty: paths.*.*.operationId
    value: [myPutEndpoint,myPostEndpoint]

For my part, I've already set up a workaround via script, but I'will see the use of custom decorator for industrialization.

**Comment by tatomyr** [comment]
@arnaudroubinet feel free to ping me if you need any help with that.

**Comment by tatomyr** [comment]
Out of curiosity, would you find it helpful to have a similar decorator that could directly target paths and operations that you would like to pick up? Something like the following:

**redocly.yaml**
```yaml
decorators:
  pick:
  -  /test:
    - put
    - post
```

**Comment by arnaudroubinet** [comment]
Clearly it would do the job.

**Comment by arnaudroubinet** [comment]
The difficulty lies in restricting this filter to paths and not altering the structure of the schemas nodes.

**Comment by tatomyr** [comment]
> The difficulty lies in restricting this filter to paths and not altering the structure of the schemas nodes.

Yes, you probably should use the [PathItem](https://redocly.com/learn/openapi/openapi-visual-reference/path-item) visitor in the decorator and operate on that level (decide what operations to keep and what to remove).
